nats/src/main.rs

```diff
@@ -25,40 +29,44 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     // handle lattice control messages and forward rpc to the provider dispatch
     // returns when provider receives a shutdown control message
     let host_data = load_host_data()?;
-    let provider = if let Some(c) = host_data.config_json.as_ref() {
-        let config: ConnectionConfig = serde_json::from_str(c)
-            .expect("JSON deserialization from connection config should have worked");
-        NatsMessagingProvider {
-            default_config: config,
-            ..Default::default()
-        }
-    } else {
-        NatsMessagingProvider::default()
-    };
+    let provider = generate_provider(host_data);
     provider_main(provider, Some("Nats Messaging Provider".to_string()))?;
 
     eprintln!("Nats-messaging provider exiting");
     Ok(())
 }
 
-type NatsTopic = String;
-type ClusterUri = String;
-type AuthJwt = String;
-type AuthSeed = String;
+fn generate_provider(host_data: HostData) -> NatsMessagingProvider {
+    if let Some(c) = host_data.config_json.as_ref() {
```

Comment:
Nit: This would probably be better as a `match` block with an `if` guard, but not gonna block on it